### PR TITLE
Add support for image sequence data (e.g. calcium imaging of neuronal activity)

### PR DESCRIPTION
**Pull request by apdavison**
provides the classes `ImageSequence` and three `RegionOfInterest` subclasses, as well as three IO modules for imaging data.

The principal author of this PR was @teogale

### CODE REVIEW

**Comment by samuelgarcia** [comment]
Hi Andrew
it is a big one.
I will have a look.

File: neo/core/imagesequence.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+    def __array_finalize__spec(self, obj):
+
+        self.sampling_rate = getattr(obj, 'sampling_rate', None)
+        self.spatial_scale = getattr(obj, 'spatial_scale', None)
+        self.units = getattr(obj, 'units', None)
+
+        return obj
+
+    def signal_from_region(self, *region):
```
Comment (samuelgarcia) [inline neo/core/imagesequence.py]: I think we would a **reduce_mode='average' / 'median'/'max'** kargs to control the method of the reduce.
No ?

File: neo/core/regionofinterest.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+
+from math import floor, ceil
+
```
Comment (samuelgarcia) [inline neo/core/regionofinterest.py]: A general and stupid question : this RegionOfInterest is binary by design (in or out) but we could also have a weighted region of interest. No ?

File: neo/core/regionofinterest.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+                max_y = i[1]
+
+        list_coord = []
+        for y in range(int(floor(min_y)), int(ceil(max_y))):
+            for x in range(int(floor(min_x)), int(ceil(max_x))):
+                list_coord.append((x, y))
+
+        pixel_list = self.polygon_ray_casting(self.vertices, list_coord)
+
+        return pixel_list
```
Comment (samuelgarcia) [inline neo/core/regionofinterest.py]: We could also have a BooleanMaskRegionOfInterest, no ?

File: neo/core/imagesequence.py (hunk 1 above)
Comment (apdavison) [reply neo/core/imagesequence.py]: yes, that would be a good improvement. I suggest leaving it for a future update, however.

File: neo/core/regionofinterest.py (hunk 2 above)
Comment (apdavison) [reply neo/core/regionofinterest.py]: Yes, that would make sense for circular regions of interest (e.g. circular gaussian weighting). I'm not sure how you would specify it for a more complex shape, however.

File: neo/core/regionofinterest.py (hunk 3 above)
Comment (apdavison) [reply neo/core/regionofinterest.py]: Yes, we should add that in a future version.

**Comment by apdavison** [comment]
ok to merge?

**Comment by apdavison** [comment]
> I am not super happy how the raltionship between ROI and block is done.

I'm not happy either, but, as you say, we need to refactor relationships anyway!

Thanks for the merge.

**Pull request merged**